integration/local_resource_test.go

```diff
@@ -1,34 +1,52 @@
-//+build integration
+// +build integration
 
 package integration
 
 import (
 	"io/ioutil"
 	"os"
+	"strings"
 	"syscall"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 const cleanupTxt = "cleanup.txt"
 
-func TestLocalResourceCleanup(t *testing.T) {
+func TestLocalResource(t *testing.T) {
 	f := newFixture(t, "local_resource")
 	defer f.TearDown()
 
-	defer func() {
+	t.Cleanup(func() {
 		_ = os.Remove(f.testDirPath(cleanupTxt))
-	}()
+		_ = os.Remove(f.testDirPath("probe-success"))
+	})
 
 	f.TiltWatch()
 
+	const barServeLogMessage = "Running serve cmd: ./hello.sh bar"
+	const readinessProbeSuccessMessage = `"[readiness probe] fake probe success message"`
+
 	require.NoError(t, f.logs.WaitUntilContains("hello! foo #1", 5*time.Second))
+
+	// write a sentinel file for the probe to find and change its result
+	require.NoError(t, ioutil.WriteFile(f.testDirPath("probe-success"), nil, 0777))
+	require.NoError(t, f.logs.WaitUntilContains("[readiness probe] fake probe success message", 5*time.Second))
+
+	// wait for second resource to start and then ensure that the order in the logs is as expected
+	require.NoError(t, f.logs.WaitUntilContains(barServeLogMessage, 5*time.Second))
+	curLogs := f.logs.String()
+	assert.Greater(t, strings.Index(curLogs, barServeLogMessage), strings.Index(curLogs, readinessProbeSuccessMessage),
```

Comment:
IIUC, `readinessProbeSuccessMessage`'s `"`s don't actually appear in the logs, so `strings.Index` here returns `-1`, so the index of `barServeLogMessage` will always be greater (presuming it actually appears in the log)

It'd probably be safest to get the indexes separately and assert they're not -1, but just using `readinessProbeSuccessMessage` on line 37 would suffice (to catch the issue, and then you'd need to remove the quotes from the const).

Reply:
Whoops - that was the whole reason I'd made a `const` with the log message so that I could assert they both existed before doing index magic but did it very sloppily 😅

Reply:
Fixed in fd7cd65d